source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -123,26 +123,33 @@ def _extend_pt_sections_with_fallback_sections(pt_journey, dp_journey):
         pt_journey.sections.extend(dp_journey.journeys[0].sections)
 
 
-def _update_fallback_sections(pt_journey, fallback_dp, fallback_period_extremity):
+def _update_fallback_sections(journey, fallback_dp, fallback_period_extremity, fallback_type):
     """
-    Replace pt_journey's fallback sections with the given fallback_dp.
+    Replace journey's fallback sections with the given fallback_dp.
 
-    Note: the replacement is done in place of the pt_journey
+    Note: the replacement is done in place of the journey
     """
     aligned_fallback = _align_fallback_direct_path_datetime(fallback_dp, fallback_period_extremity)
     fallback_sections = aligned_fallback.journeys[0].sections
 
     # update the 'id' which isn't set
     _rename_fallback_sections_ids(fallback_sections)
 
-    if fallback_period_extremity.represents_start:
-        section_to_update = pt_journey.sections[-1]
+    if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK:
+        section_to_replace = journey.sections[0]
     else:
-        section_to_update = pt_journey.sections[0]
+        section_to_replace = journey.sections[-1]
+
+    journey.sections.remove(section_to_replace)
 
-    pt_journey.sections.remove(section_to_update)
-    pt_journey.sections.extend(fallback_sections)
-    pt_journey.sections.sort(SectionSorter())
+    # We have to create the link between the fallback and the pt part manually here
+    if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK:
+        fallback_sections[0].destination.CopyFrom(journey.sections[0].origin)
```

Comment:
Need `fallback_sections[-1]` as a direct path can be several sections (parking, BSS...). Testing this could be great.